### PR TITLE
use DiscoverInstances

### PR DESCRIPTION
**Pull request by hanshasselberg**
DiscoverInstances is only useable when it is a HTTPNamespaces that is being synced.

### CODE REVIEW

**Comment by herrhound** [comment]
i0rek, just a note – DiscoverInstances is available with every single AWS Cloud Map namespace. This is the default discovery mode.

**Comment by hanshasselberg** [comment]
@herrhound thanks for reaching out! I tried that, but couldn't get it to work, but I will try it again.

**Comment by hanshasselberg** [comment]
@herrhound of course, you are right :). Must have been an issues with the namespace I was using for testing, recreated and now it works.

File: catalog/aws.go [hunk 1]
```diff
@@ -95,23 +112,47 @@ func (a *aws) transformServices(aservices []sd.ServiceSummary) map[string]servic
 	return services
 }
 
+func (a *aws) transformNamespace(anamespace *sd.Namespace) namespace {
+	namespace := namespace{id: *anamespace.Id, name: *anamespace.Name}
```
Comment (banks) [inline catalog/aws.go]: nit: I find the variable names here kinda hard to follow on first reading! `anamespace` I thought was a typo at first until I realised it's trying to avoid colliding with the internal `namespace`. Although it's technically not a problem I also found it hard to parse at first that `namespace` is both a variable and a type name in this line.

I'd be tempted to call the argument `ns` and the internal one `n` or something. Kinda arbitrary but easier to distinguish within this short method IMO.

**Review by banks — APPROVED** [review]
Some minor nits inline for the code. But I don't see any obvious errors. 

That said I don't really know the code here nor the AWS APIs you're using so I can't really say there are no bad assumptions or bugs lurking! Tests look good though.

File: catalog/aws.go (hunk 1 above)
Comment (hanshasselberg) [reply catalog/aws.go]: `anamespace` is also a shorthand for `awsnamespace` because it is the one coming from the sdk in contrast to the other one.

Comment (hanshasselberg) [reply catalog/aws.go]: I changed it to `awsNamespace` here and in other places too.
